index/index.go

```diff
@@ -143,7 +143,11 @@ func NewWriter(fn string) (*Writer, error) {
 	if err != nil {
 		return nil, err
 	}
-	defer df.Close() // close for flatform windows
+	defer df.Close() // Close for platform windows.
+
+	if err := os.RemoveAll(fn); err != nil {
+		return nil, errors.Wrap(err, "remove any existing index at path")
+	}
```

Comment:
Mh, do we need this? Just opening a new file in create/write mode on it should just overwrite it as expected.

Reply:
Yep. So if there happened to be an index that is larger than the one being written it'll cause issues. Like one of my tests, I wrote a bigger one and then a smaller one, but couldn't figure out what's wrong.

Very rare edge case, but I guess it doesn't hurt to add a check.